### DISCUSSION TITLE
Bash condition not working

### DISCUSSION THREAD

**Discussion by MostefaKamalLala** [community/community#27107] [Actions]
<p>Hi,<br>
I have a simple bash condition which works properly on my local terminal and another online terminal. However for some reason, it’s not working properly on github runner running on ubuntu-latest.</p>
<p>Here is the step:</p>
<pre><code class="lang-yaml">    - name: Perform action
      shell: bash
      run: |  
          cluster_status=$(az aks list | jq -c 'map(select(any(.name; contains("dev-aks")))|.powerState)[].code')
          echo $cluster_status
          if [[ "$cluster_status" == "Stopped" &amp;&amp; "${{ steps.set-action.outputs.action }}" == "stop" ]]; then
             echo 'cluster already stopped, exiting'
          elif [[ "$cluster_status" == "Started" &amp;&amp; "${{ steps.set-action.outputs.action }}" == "start"  ]]; then 
              echo 'cluster already started, exiting'
          else
              az aks ${{ steps.set-action.outputs.action }} --name dev-aks --resource-group dev-rg
          fi
</code></pre>
<p>I echoed both variables in the condition:<br>
${{ steps.set-action.outputs.action }} = stop<br>
$cluster_status = Stopped</p>
<p>but I still getting the last command in the else executed. I even tried with</p>
<pre><code class="lang-auto">  if [[ "$cluster_status" == "Stopped" ]]; then
</code></pre>
<p>and it still doesn’t get in!</p>
<p>Bash version is 5.0.17(1)-release.</p>

**Comment by airtower-luna** [comment]
<p>I’m not sure if that’s what’s causing the problem, but is there any particular reasons why you’re defining <code>cluster_status</code> as an array?</p>

**Comment by MostefaKamalLala** [comment, marked as answer]
<p>I did define differently before it’s a copy paste of my last try lol. <a class="mention" href="/airtower-luna">@airtower-luna</a> .</p>
<p>I finally made it work. For anyone who gets the same problem here is the solution:<br>
We need to create a step before using the value of cluster_status and set it in an evironnement variable.<br>
Then, I don’t know why and I hope someone clarifies this for us, it works by using it in another step.</p>
<pre><code class="lang-auto">    - name: Get AKS status
      run: |
          echo "cluster_status=$(az aks list | jq -c 'map(select(any(.name; contains("dev-aks")))|.powerState)[].code')" &gt;&gt; $GITHUB_ENV         

    - name: Perform action
      run: |    
          if [[ ${{ env.cluster_status }} == "Stopped" &amp;&amp; "${{ steps.set-action.outputs.action }}" == "stop" ]]; then
             echo 'cluster already stopped, exiting'
          elif [[ ${{ env.cluster_status }} == "Started" &amp;&amp; "${{ steps.set-action.outputs.action }}" == "start"  ]]; then 
              echo 'cluster already stopped, exiting'
          else
              az aks ${{ steps.set-action.outputs.action }} --name dev-aks --resource-group dev-rg
          fi  
</code></pre>
